core/src/main/java/org/apache/iceberg/hadoop/HadoopTables.java

```diff
@@ -144,6 +147,52 @@ public Table create(Schema schema, PartitionSpec spec, SortOrder order,
     return new BaseTable(ops, location);
   }
 
+  /**
+   * Drop a table and delete all data and metadata files. Throws NoSuchTableException if the table does not exists.
+   *
+   * @param location a path URI (e.g. hdfs:///warehouse/my_table)
+   * @return true if the table was dropped
+   */
+  public boolean dropTable(String location) {
```

Comment:
I think this wasn't implemented before because it is not part of the `Tables` API, but now that this is the only implementation, maybe we should consider just deprecating the `Tables` API and making `HadoopTables` a stand-alone class.

Reply:
Maybe this would merit another discussion, and another PR

Reply:
Agreed.